### PR TITLE
Linear Regression in C++. #733

### PR DESCRIPTION
**Pull request by aayushbest**
**Hi!**  
Thanks for considering for contributing to this ever-growing list of algorithm and data structure implementations.
Your contribution is valuable. Use the follow template for creating PRs. (Delete this part)
---

**Fixes #{ISSUE_NUMBER}**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [ ] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [ ] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [ ] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [ ] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [ ] This pull request shall only be merged once the build passes.

-   [ ] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-lreg/cpp/linear_regression.cpp [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  double calculateBias() const;
+  // Mutators
+  void setWeight(const double weight);
+  void setBias(const double bias);
+};
+/**
+ *  LinearRegression Interface implementation.
+ */
+template<typename T, typename U>
+LinearRegression<T,U>::LinearRegression(const vector<DataPoint<T, U>> data){
```
Comment (RavinduSachintha) [inline algorithms/ar-lreg/cpp/linear_regression.cpp:95]: Codacy found an issue: [Member variable 'LinearRegression < int , int >::weight' is not initialized in the constructor.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6399935)

Comment (RavinduSachintha) [inline algorithms/ar-lreg/cpp/linear_regression.cpp:95]: Codacy found an issue: [Function parameter 'data' should be passed by const reference.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6399935)

File: algorithms/ar-lreg/cpp/linear_regression.cpp [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    dataset = data;
```
Comment (RavinduSachintha) [inline algorithms/ar-lreg/cpp/linear_regression.cpp:96]: Codacy found an issue: [Variable 'dataset' is assigned in constructor body. Consider performing initialization in initialization list.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6399935)

File: algorithms/ar-lreg/cpp/linear_regression.cpp [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+};
+/**
+ *  LinearRegression Interface implementation.
+ */
+
+template<typename T, typename U>
+LinearRegression<T,U>::LinearRegression():weight(0.0),bias(0.0){}
+
+template<typename T, typename U>
+LinearRegression<T,U>::LinearRegression(const vector<DataPoint<T, U>> &data): dataset(data){}
```
Comment (RavinduSachintha) [inline algorithms/ar-lreg/cpp/linear_regression.cpp:100]: Codacy found an issue: [Member variable 'LinearRegression < int , int >::weight' is not initialized in the constructor.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6399935)

**Comment by Danushka96** [comment]
Hi @aayushbest 
Can you update the readme file?

**Comment by aayushbest** [comment]
Sure

**Review by Danushka96 — APPROVED** [review]

**Comment by aayushbest** [comment]
Hi @Danushka96 
Can you assign me some issue in Java on some data structure or algorithm except heapsort, knn or linear regression?
Eager to work :)

**Comment by Danushka96** [comment]
Hi @aayushbest 
check the algorithm table and let me know what you can do,

**Comment by aayushbest** [comment]
Hi @Danushka96 
I can do Breadth First Search in Java.

**Comment by Danushka96** [comment]
@aayushbest 
It's already there right?

**Comment by aayushbest** [comment]
@Danushka96 
Yes. Later I found it is added recently but the README was not updated.

**Pull request merged**